busylight/api/busylight_api.py

```diff
@@ -73,6 +87,22 @@ def __init__(self):
             self.username = None
             self.password = None
 
+        logger.info(
+            "Set up CORS Access-Control-Allow-Origin header, if environment variable is set.")
+        try:
+            self.origins = json_loads(
+                environ["BUSYLIGHT_API_CORS_ORIGINS_LIST"])
```

Comment:
Again, we can supply a default value for "BUSYLIGHT_API_CORS_ORIGINS_LIST" and avoid having to handle the exception:
```python
self.origins = json_loads(environ.get("BUSYLIGHT_API_CORS_ORIGINS_LIST", [])
```

I prefer this since it simplifies how we operate on `self.origins` in the following code. We should also do some checking to make sure that `self.origins` is what we expect it to be (presumably a list of strings?) so that we don't accidentally parse some unfriendly stuff from the user (intentionally bad or otherwise).

Reply:
I agree. I'm not sure how to do the checking, but I agree 😅 . Perhaps a type-check to see if it's a `list()`.

Reply:
Something like:
```
if not isinstance(self.origins, list) or any(not isinstance(item, str) for item in self.origins):
      # some kind of error noise via logging and clear `self.origins` to a known safe value (empty list?)
```

Reply:
I'm considering using `logger.warning()` -- should I use `logger.error()` instead (you did mention `error noise`)?

Reply:
A warning is perfectly acceptable, I meant it more along the lines of ["make a jazz noise here"](https://en.wikipedia.org/wiki/Make_a_Jazz_Noise_Here) which is probably pretty obscure now that I have to explain it.

Reply:
Ah, no worries! It's now a warning with value-reset logic attached. :)